src/main/java/io/confluent/connect/elasticsearch/Validator.java

```diff
@@ -62,12 +74,22 @@ public Validator(Map<String, String> props) {
     values = validations.stream().collect(Collectors.toMap(ConfigValue::name, Function.identity()));
   }
 
+  // Exposed for testing
+  protected Validator(Map<String, String> props, RestHighLevelClient client) {
+    this(props);
+    this.client = client;
+  }
+
   public Config validate() {
     if (config == null) {
       // individual configs are invalid, no point in validating combinations
       return new Config(validations);
     }
 
+    if (client == null) {
+      client = createClient();
```

Comment:
Should it be made `final` and guarded by a `synchronized` block?

Reply:
Ok, I see a logic below, and not sure why we even have the `client` as a class field vs just have it as a local variable, since we are not reusing it between calls?

Reply:
for testing. there is no clean way that I have found to mock the client, open to suggestions

Reply:
```
protected Validator(Map<String, String> props, ClientFactory clientFactory) {}
```

```
private RestHighLevelClient createClient() {
if this.clientFactory != null {
return clientFactory.create();
} else {
//use your default factory
}}
```

```
interface ClientFactory {
RestHighLevelClient create();
}
```

Reply:
Then in your test, just pass a lambda function that creates a mocked client to `Validator` constructor.